pkg/occlient/monitor.go

```diff
@@ -0,0 +1,171 @@
+	}
+	defer deploymentConfigWatcher.Stop()
+
+	for {
+
+		// Wait for some information
+		dcVal, ok := <-deploymentConfigWatcher.ResultChan()
+		if !ok {
+			deploymentConfigErrorChannel <- errors.New("watch channel was closed")
+			break
+		}
+
+		if e, ok := dcVal.Object.(*appsv1.DeploymentConfig); ok {
+
+			// Output some statuses / current information
+			deploymentConfigSpinner.Status(fmt.Sprintf("DeploymentConfig: Requested Replicas (%d) | Available Replicas (%d)", e.Status.Replicas, e.Status.AvailableReplicas))
+
+			// If we're successful
+			if e.Status.Replicas > 0 && (e.Status.Replicas == e.Status.AvailableReplicas) {
+				deploymentConfigSpinner.Success()
+				wg.Done()
+				break
+			}
+		} else {
+			deploymentConfigErrorChannel <- errors.New("unable to convert event object to DeploymentConfig")
+			break
+		}
+	}
+
+}
+
+// waitForPod waits for the pod to successfully complete / come up
+func (c *Client) waitForPod(podSelector string, podSpinner *fidget.Spinny) {
+
+	// Create the watcher
+	podWatcher, err := c.kubeClient.CoreV1().Pods(c.Namespace).Watch(metav1.ListOptions{
+		LabelSelector: podSelector,
+	})
+	if err != nil {
+		podErrorChannel <- errors.Wrapf(err, "unable to watch pod")
+	}
+
+	// This will only happen if the wrong "name" was passed in, or you'll get an invalid memory address / nil pointer reference
+	if podWatcher == nil {
+		podErrorChannel <- errors.Errorf("unable to retrieve any pod by the selector %s", podSelector)
+	}
+	defer podWatcher.Stop()
+
+	for {
+
+		// Wait for something to happen
+		val, ok := <-podWatcher.ResultChan()
+		if !ok {
+			podErrorChannel <- errors.New("watch channel was closed")
+			break
+		}
+
+		// Evaluate the channel / object you retrieved
+		if e, ok := val.Object.(*corev1.Pod); ok {
+
+			// Output some status information
+			podSpinner.Status(fmt.Sprintf("Pod: %s", e.Status.Phase))
+
+			switch e.Status.Phase {
+
+			// If we're successful
+			case corev1.PodRunning:
+				podSpinner.Success()
+				wg.Done()
+
+			// If we have failed, we've got to output some more verbose output..
+			case corev1.PodFailed, corev1.PodUnknown:
+				podSpinner.Fail()
+				podErrorChannel <- errors.Errorf("pod %s status %s", e.Name, e.Status.Phase)
+			}
+
+		} else {
+			podErrorChannel <- errors.New("unable to convert event object to Pod")
+			break
+		}
+	}
+	podErrorChannel <- errors.New("unable to convert event object to Pod")
+}
+
+// WaitForEverything will wait for EVERYTHING to be RUNNING! (replacement for "WaitAndGetPod" since this far more verbose
+// and elaborate..
+// TODO:
+// Use `kubectl get events -w` for LAST occuring event? Also, we shouldn't really be displaying this if we're redeploying? Else new
+// PVC's need to be created, etc.
+// For --git build should we output the same stuff? Got to output builconfig too?
+// Secrets
+// PVC?
+// buildconfig?
+func (c *Client) WaitForEverything(podSelector string, name string, timeout time.Duration) error {
```

Comment:
what is `name`? - name of what?

Reply:
If name is DeploymentConfig name, than you don't need `podSelector` as you can get it from DeploymentConfig.